# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../partimage/partimage-0.6.9-zlib-1.2.5.2.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

Fix building with zlib >= 1.2.5.2

http://bugs.gentoo.org/405323

--- partimage-0.6.9-orig/src/client/imagefile.cpp
+++ partimage-0.6.9/src/client/imagefile.cpp
@@ -783,7 +783,7 @@
   else if (m_options.dwCompression == COMPRESS_GZIP) // Gzip compression
     {
       showDebug(1, "open gzip\n");
-      m_gzImageFile = (gzFile *) gzdopen(m_nFdImage, "wb"); //"wb1h");
+      m_gzImageFile = gzdopen(m_nFdImage, "wb"); //"wb1h");
       if (m_gzImageFile == NULL)
 	{
 	  showDebug(1, "error:%d %s\n", errno, strerror(errno));
@@ -1098,7 +1098,7 @@
     }
   else if (m_options.dwCompression == COMPRESS_GZIP) // Gzip compression
     {
-      m_gzImageFile = (gzFile *) gzdopen(m_nFdImage, "rb");
+      m_gzImageFile = gzdopen(m_nFdImage, "rb");
       if (m_gzImageFile == NULL)
         THROW(ERR_ERRNO, errno);
       else
--- partimage-0.6.9-orig/src/client/imagefile.h
+++ partimage-0.6.9/src/client/imagefile.h
@@ -41,7 +41,7 @@
   COptions m_options;
 
   FILE *m_fImageFile;
-  gzFile *m_gzImageFile;
+  gzFile m_gzImageFile;
   BZFILE *m_bzImageFile;
 
   int m_nFdImage;
